bytecomp/bytegen.ml

```diff
@@ -505,6 +505,38 @@ let comp_primitive stack_info p sz args =
     ->
       fatal_error "Bytegen.comp_primitive"
 
+let insert_hint p cont =
+  match p with
+  | Pmakeblock (_, Immutable, _) | Pmakearray (_, Immutable) ->
+      Khint Hint_immutable :: cont
+  | Pbigarrayref(unsafe, _, elt_kind, layout)
+  | Pbigarrayset (unsafe, _, elt_kind, layout) ->
+      Khint (Hint_bigarray {unsafe; elt_kind; layout}) :: cont
+  | Pstring_load_16 true
+  | Pstring_load_32 true
+  | Pstring_load_64 true
+  | Pbytes_set_16 true
+  | Pbytes_set_32 true
+  | Pbytes_set_64 true
+  | Pbytes_load_16 true
+  | Pbytes_load_32 true
+  | Pbytes_load_64 true
+  | Pbigstring_load_16 true
+  | Pbigstring_load_32 true
+  | Pbigstring_load_64 true
+  | Pbigstring_set_16 true
+  | Pbigstring_set_32 true
+  | Pbigstring_set_64 true ->
+      Khint (Hint_unsafe) :: cont
+  | Parraylength kind ->
+      Khint (Hint_array kind) :: cont
```

Comment:
Why a hint for Parray_length and not for Parray_(ref/set)(s,u) ?

Reply:
The bytecode compiler generates different code for Parray_(ref/set)(s,u) depending on the array kind.